usl_models/trainer/flood_task.py

```diff
@@ -0,0 +1,133 @@
+import argparse
+import logging
+import os
+
+from google.cloud import firestore  # type:ignore[attr-defined]
+from google.cloud import storage  # type:ignore[attr-defined]
+import tensorflow as tf
+from tensorflow.python.client import device_lib
+
+import usl_models.flood_ml.dataset
+import usl_models.flood_ml.model
+import usl_models.flood_ml.model_params
+
+
+logging.basicConfig(level=logging.INFO)
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--epochs", dest="epochs", type=int, help="Number of epochs.")
+parser.add_argument(
+    "--batch-size", dest="batch_size", type=int, help="Size of a batch."
+)
+parser.add_argument("--sim-names", dest="sim_names", nargs="+", type=str, required=True)
+
+parser.add_argument(
+    "--model-dir",
+    dest="model_dir",
+    default=os.environ.get("AIP_MODEL_DIR"),
+    type=str,
+    help="Model dir.",
+)
+parser.add_argument(
+    "--distribute",
+    dest="distribute",
+    type=str,
+    default="single",
+    help="distributed training strategy",
+    choices=["single", "mirrored", "multiworker", "tpu"],
+)
+
+args = parser.parse_args()
+
+logging.info("DEVICES" + str(device_lib.list_local_devices()))
+
+# Single Machine, single compute device
+if args.distribute == "single":
+    if tf.test.is_gpu_available():
+        strategy = tf.distribute.OneDeviceStrategy(device="/gpu:0")
+    else:
+        strategy = tf.distribute.OneDeviceStrategy(device="/cpu:0")
+    logging.info("Single device training")
+# Single Machine, multiple compute device
+elif args.distribute == "mirrored":
+    strategy = tf.distribute.MirroredStrategy()
```

Comment:
Have we tried this strategy yet? If this works we should definitely use it for the larger training job, we could support a larger batch size.

Reply:
I have tried none of these strategies -- this is stuff I copy-pasted from vertex documentation but haven't actually tried.